src/Nethermind/Nethermind.Network.Discovery/NettyDiscoveryHandler.cs

```diff
@@ -183,16 +182,43 @@ private byte[] Serialize(DiscoveryMsg msg)
     {
         return msg.MsgType switch
         {
-            MsgType.Ping => _msgSerializationService.Serialize((PingMsg) msg),
-            MsgType.Pong => _msgSerializationService.Serialize((PongMsg) msg),
-            MsgType.FindNode => _msgSerializationService.Serialize((FindNodeMsg) msg),
-            MsgType.Neighbors => _msgSerializationService.Serialize((NeighborsMsg) msg),
-            MsgType.EnrRequest => _msgSerializationService.Serialize((EnrRequestMsg) msg),
-            MsgType.EnrResponse => _msgSerializationService.Serialize((EnrResponseMsg) msg),
+            MsgType.Ping => _msgSerializationService.ZeroSerialize((PingMsg) msg).ReadAllBytes(),
```

Comment:
Please add a comment why we want to use ZeroSerialize instead of Serialize

Reply:
There is no serialize method now, only ZeroSerialize. Thinking of renaming ZeroSerialize -> Serialize